tikv-client/src/main/java/com/pingcap/tikv/expression/Constant.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv.expression;
+
+import com.google.common.collect.ImmutableList;
+import com.pingcap.tikv.exception.TiExpressionException;
+import com.pingcap.tikv.types.DataType;
+import com.pingcap.tikv.types.DateTimeType;
+import com.pingcap.tikv.types.DateType;
+import com.pingcap.tikv.types.DecimalType;
+import com.pingcap.tikv.types.IntegerType;
+import com.pingcap.tikv.types.RealType;
+import com.pingcap.tikv.types.StringType;
+import com.pingcap.tikv.types.TimestampType;
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Objects;
+import org.joda.time.DateTime;
+
+
+// Refactor needed.
+// Refer to https://github.com/pingcap/tipb/blob/master/go-tipb/expression.pb.go
+// TODO: This might need a refactor to accept an DataType?
+public class Constant implements Expression {
+  private final Object value;
+  private DataType type;
+
+  public static Constant create(Object value, DataType type) {
+    return new Constant(value, type);
+  }
+
+  public static Constant create(Object value) {
+    return new Constant(value, null);
+  }
+
+  public Constant(Object value, DataType type) {
+    this.value = value;
+    this.type = type == null ? getDefaultType(value) : type;
+  }
+
+  protected static boolean isIntegerType(Object value) {
+    return value instanceof Long
+        || value instanceof Integer
+        || value instanceof Short
+        || value instanceof Byte;
```

Comment:
Is it necessary to consider `BigInteger` here?

Reply:
Seems not necessary for me since it looks like no column matches to BigInteger for now. If you need a big number you can use BigDecimal.